iis-wf/iis-wf-referenceextraction/src/test/java/eu/dnetlib/iis/wf/referenceextraction/patent/PatentFacadeMock.java

```diff
@@ -0,0 +1,39 @@
+package eu.dnetlib.iis.wf.referenceextraction.patent;
+
+import java.util.NoSuchElementException;
+
+import eu.dnetlib.iis.referenceextraction.patent.schemas.ImportedPatent;
+
+/**
+ * Simple mock which provides concatenation of all {@link ImportedPatent} fields as generated meta.
+ * 
+ * @author mhorst
+ *
+ */
+public class PatentFacadeMock implements PatentServiceFacade {
+
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public String getPatentMetadata(ImportedPatent patent) throws NoSuchElementException, Exception {
+        return generateMeta(patent);
+    }
+
+    protected static String generateMeta(ImportedPatent patent) {
+        if ("non-existing".equals(patent.getPublnNr())) {
```

Comment:
IDE says: 'String.equals()' called with 'CharSequence' argument. Wants to convert it to use `contentEquals`.

Reply:
That's a good catch. 

I've slipped on a similar problem once when comparing String with an object which turned out to be an `Utf8` class instance.

In this particular case it worked properly because `patent.getPublnNr()` was nice enough to materialize itself as a `String` :)